cpp/include/Ice/SSL.h

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#ifndef ICE_SSL_H
+#define ICE_SSL_H
+
+#include "SSLConnectionInfo.h"
+
+#include <functional>
+
+#ifdef _WIN32
+// We need to include windows.h before wincrypt.h.
+// clang-format off
+#    ifndef NOMINMAX
+#        define NOMINMAX
+#    endif
+#    include <windows.h>
+#    include <wincrypt.h>
+// clang-format on
+// SECURITY_WIN32 or SECURITY_KERNEL, must be defined before including security.h indicating who is compiling the code.
+#    ifdef SECURITY_WIN32
+#        undef SECURITY_WIN32
+#    endif
+#    ifdef SECURITY_KERNEL
+#        undef SECURITY_KERNEL
+#    endif
+#    define SECURITY_WIN32 1
+#    include <schannel.h>
+#    include <security.h>
+#    include <sspi.h>
+#    undef SECURITY_WIN32
+#endif
+
+namespace Ice::SSL
```

Comment:
Is the idea to rename IceSSL -> Ice::SSL in another PR?

Reply:
Yes, and do the same in C#/Java replacing the top-level IceSSL namespace with Ice::SSL.